packages/web-components/@sfx/sayt/test/unit/sayt.test.ts

```diff
@@ -222,6 +229,120 @@ describe('Sayt Component', () => {
     });
   });
 
+  describe('requestSayt()', () => {
+    let dispatchEvent, query, searchbox;
+    beforeEach(() => {
+      query = 'some-query';
+      searchbox = 'some-searchbox-id';
+      sayt.minSearchLength = 3;
+      dispatchEvent = stub(window, 'dispatchEvent');
+    });
+
+    it('should not dispatch an event if query length is sufficiently short', () => {
+      query = 'ab';
+
+      sayt.requestSayt(query, searchbox);
+
+      expect(dispatchEvent).to.not.be.called;
+    });
+
+    it('should request sayt autocomplete terms and products', () => {
+      const requestSaytAutocompleteTerms = stub(sayt, 'requestSaytAutocompleteTerms');
+      const requestSaytProducts = stub(sayt, 'requestSaytProducts');
+
+      sayt.requestSayt(query, searchbox);
+
+      expect(requestSaytAutocompleteTerms).to.be.calledWith(query, searchbox);
+      expect(requestSaytProducts).to.be.calledWith(query, searchbox);
+    });
+  });
+
+  describe('handleAutocompleteTermHover', () => {
+    it('should call requestSaytProducts() with the event query and the searchbox', () => {
+      const requestSaytProducts = stub(sayt, 'requestSaytProducts');
+      const query = 'some-query';
+      const event = { detail: { query } };
+      const searchbox = sayt.searchbox = 'some-searchbox-id';
+
+      sayt.handleAutocompleteTermHover(event);
+
+      expect(requestSaytProducts).to.be.calledWith(query, searchbox);
+    });
+  });
+
+  describe('requestSaytAutocompleteTerms', () => {
+    let dispatchEvent, query, searchbox;
+    beforeEach(() => {
+      query = 'some-query';
+      searchbox = 'some-searchbox-id';
+      dispatchEvent = stub(window, 'dispatchEvent');
+    });
+
+    it('should dispatch an event with a payload that includes the query and searchbox', () => {
+      const eventObj = {};
+      const CustomEvent = stub(window, 'CustomEvent').returns(eventObj);
+
+      sayt.requestSaytAutocompleteTerms(query, searchbox);
+
+      expect(CustomEvent).to.be.calledWith('sfx::autocomplete_fetch_data', {
```

Comment:
Tests for `requestSaytAutocompleteTerms()` and `requestSaytProducts()` should reference the correct exported events.